python/lsst/daf/butler/core/datasets.py

```diff
@@ -68,24 +80,11 @@ def template(self):
         """
         return self._template
 
-    @property
-    def units(self):
-        """A `DataUnitSet` that defines the `DatasetRef`\ s corresponding
-        to this `DatasetType`.
-        """
-        return self._units
-
-    @property
-    def storageClass(self):
-        """A `StorageClass` that defines how this `DatasetType` is persisted.
-        """
-        return self._storageClass
-
-    def __init__(self, name, template, units, storageClass):
+    def __init__(self, name, dataUnits, storageClass, template=None):
```

Comment:
Is the `dataUnits` argument here a `frozenset` and you copy it in the constructor?

Reply:
It may be any iterable, but it becomes a `frozenset` in the constructor.